lib/workload/stateless/postgres_manager/construct/postgresManager.ts

```diff
@@ -0,0 +1,117 @@
+import { Duration } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as nodejs from 'aws-cdk-lib/aws-lambda-nodejs';
+import * as secretsmanager from 'aws-cdk-lib/aws-secretsmanager';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as rds from 'aws-cdk-lib/aws-rds';
+import * as ssm from 'aws-cdk-lib/aws-ssm';
+import { DbAuthType, MicroserviceConfig } from '../function/utils';
+
+export type PostgresManagerConfig = {
+  masterSecretName: string;
+  dbClusterIdentifier: string;
+  microserviceDbConfig: MicroserviceConfig;
+  clusterResourceIdParameterName: string;
+};
+
+export type PostgresManagerStackProps = PostgresManagerConfig & {
+  vpc: ec2.IVpc;
+  lambdaSecurityGroup: ec2.ISecurityGroup;
+};
+
+export class PostgresManager extends Construct {
+  constructor(scope: Construct, id: string, props: PostgresManagerStackProps) {
+    super(scope, id);
+
+    const { dbClusterIdentifier, microserviceDbConfig } = props;
+
+    const masterSecret = secretsmanager.Secret.fromSecretNameV2(
+      this,
+      'RdsMasterSecret',
+      props.masterSecretName
+    );
+
+    const dbClusterResourceId = ssm.StringParameter.valueFromLookup(
+      this,
+      '/orcabus/db-cluster-resource-id'
+    );
+
+    const rdsLambdaProps = {
+      timeout: Duration.minutes(5),
+      depsLockFilePath: __dirname + '/../yarn.lock',
+      handler: 'handler',
+      runtime: lambda.Runtime.NODEJS_20_X,
+      architecture: lambda.Architecture.ARM_64,
+      environment: {
+        RDS_SECRET_MANAGER_NAME: masterSecret.secretName,
+        MICROSERVICE_CONFIG: JSON.stringify(props.microserviceDbConfig),
+      },
+      vpc: props.vpc,
+      vpcSubnets: {
+        subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS,
+      },
+      securityGroups: [props.lambdaSecurityGroup],
+    };
+
+    // 1. lambda responsible on db creation
+    const createPgDb = new nodejs.NodejsFunction(this, 'CreateDbPostgresLambda', {
+      ...rdsLambdaProps,
+      entry: __dirname + '/../function/create-pg-db.ts',
+      functionName: 'orcabus-create-pg-db',
+    });
+    masterSecret.grantRead(createPgDb);
```

Comment:
I'm wondering if we could use this kind of pattern for database migration scripts too? I don't know if that would work though, because EdgeDb has it's own migration logic. Although, for anything using postgres directly, we could agree on some common database migration logic and re-use a Lambda function like this.

Reply:
Yes, I think it make sense just that it would probably use its own microservice login credential and perhaps the migration lambda would live under its own microservice repo/folder?

Reply:
Wouldn't you want to keep migrations (and everything that related to application specifics) together with / as part of the application?

Reply:
Yeah that's true. I guess I was just thinking in terms of generic logic that applies SQL migrations irrespective of application logic. I.e. receives as input a bunch of `.sql` files and applies the migration, recording it in a migration table.

For example, sqlx [creates](https://github.com/launchbadge/sqlx/blob/27a49914ad5e0f21b8fb1f2a62db8d838724548f/sqlx-postgres/src/migrate.rs#L117-L129) a `_sqlx_migrations` table where it has the migration version, checksum, description, success status, etc.

Reply:
Not sure if that's always a common approach across apps, but also you'd split your application concerns across multiple stacks / actors.

Reply:
Yeah that's a good point. Probably it's cleanest to leave database table creation to the actual app that is using those tables.